inc/dbmysql.class.php

```diff
@@ -786,43 +786,35 @@ function isSlave() {
     * @return boolean true if all query are successfull
     */
    function runFile($path) {
-      $DBf_handle = fopen($path, "rt");
-      if (!$DBf_handle) {
+      $script = fopen($path, 'r');
+      if (!$script) {
          return false;
       }
-
-      $formattedQuery = "";
-      $lastresult     = false;
-      while (!feof($DBf_handle)) {
-         // specify read length to be able to read long lines
-         $buffer = fgets($DBf_handle, 102400);
-
-         // do not strip comments due to problems when # in begin of a data line
-         $formattedQuery .= $buffer;
-         if ((substr(rtrim($formattedQuery), -1) == ";")
-             && (substr(rtrim($formattedQuery), -4) != "&gt;")
-             && (substr(rtrim($formattedQuery), -4) != "160;")) {
-
-            $formattedQuerytorun = $formattedQuery;
-
-            // Do not use the $DB->query
-            if ($this->query($formattedQuerytorun)) { //if no success continue to concatenate
-               $formattedQuery = "";
-               $lastresult     = true;
-               if (!isCommandLine()) {
-                  // Flush will prevent proxy to timeout as it will receive data.
-                  // Flush requires a content to be sent, so we sent sp&aces as multiple spaces
-                  // will be shown as a single one on browser.
-                  echo ' ';
-                  flush();
-               }
-            } else {
-               $lastresult = false;
+      $sql_query = @fread(
+         $script,
+         @filesize($path)
+      ) . "\n";
+      $sql_query = html_entity_decode($sql_query, ENT_COMPAT, 'UTF-8');
+
+      $sql_query = $this->removeSqlRemarks($sql_query);
+      $queries = preg_split('/;\s*$/m', $sql_query);
+
+      try {
+         $this->beginTransaction();
+         foreach ($queries as $query) {
+            $query = trim($query);
+            if ($query != '' && $query[0] != '-') {
```

Comment:
What is the purpose of `$query[0] != '-'` ?

Reply:
Lines begining with `-` are comments

Reply:
So they should be removed by call on `$this->removeSqlRemarks($sql_query);` on line 799.

Reply:
ok, done